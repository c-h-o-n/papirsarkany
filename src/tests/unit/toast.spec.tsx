import { render } from "@testing-library/react";
import { expect, test } from "vitest";
import Toast from "~/components/toast";
import type { Toast as ToastProps } from "~/lib/types";

test("Error Toast should match snapshot", () => {
  const errorToast: ToastProps = {
    id: "error-toast-test",
    type: "error",
    message: "Something terribly went wrong!",
  };

  const { container } = render(<Toast toast={errorToast} />);

  expect(container).toMatchSnapshot();
});

test("Success Toast should match snapshot", () => {
  const successToast: ToastProps = {
    id: "success-toast-test",
    type: "success",
    message: "Congrats! You are awesome!",
  };

  const { container } = render(<Toast toast={successToast} />);

  expect(container).toMatchSnapshot();
});
