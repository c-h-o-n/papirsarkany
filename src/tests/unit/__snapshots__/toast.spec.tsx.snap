// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Error Toast should match snapshot 1`] = `
<div>
  <div
    class="d-alert d-alert-error"
  >
    <span
      class="font-semibold text-white"
    >
      Something terribly went wrong!
    </span>
  </div>
</div>
`;

exports[`Success Toast should match snapshot 1`] = `
<div>
  <div
    class="d-alert d-alert-success"
  >
    <span
      class="font-semibold"
    >
      Congrats! You are awesome!
    </span>
  </div>
</div>
`;
